sdk/src/azure/platform/az_noplatform.c

```diff
@@ -2,9 +2,19 @@
 // SPDX-License-Identifier: MIT
 
 #include <azure/core/az_platform.h>
+#include <azure/core/internal/az_precondition_internal.h>
 
 #include <azure/core/_az_cfg.h>
 
-AZ_NODISCARD int64_t az_platform_clock_msec() { return 0; }
+AZ_NODISCARD az_result az_platform_clock_msec(int64_t* out_clock_msec)
+{
+  _az_PRECONDITION_NOT_NULL(out_clock_msec);
+  *out_clock_msec = 0;
+  return AZ_ERROR_DEPENDENCY_NOT_PROVIDED;
+}
 
-void az_platform_sleep_msec(int32_t milliseconds) { (void)milliseconds; }
+AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
```

Comment:
Looking at this again, I agree with Rick on the sleep API, we should consider removing `AZ_NODISCARD`.

Reply:
I don't think we should. It is us who calls these APIs. Rick's assumption is that 3 things are happening: 1. Customer calls PAL function. 2. Doesn't handle a result, and doesn't need/want to. 3. It is painful to ignore the result, which I think is not:
```
az_result ignore = az_platform_sleep_msec(1000);
(void)ignore;
```

Reply:
In that case, we should consider updating the documentation of `az_http_pipeline_policy_logging` and `az_http_pipeline_policy_retry` since they can now return any PAL-defined `az_result`. Alternatively, those APIs should take any PAL-error and convert it to a single error value, and document that.